### PR TITLE
docs: Fix stale description

### PR DESCRIPTION
**Pull request by akawashiro**
I found `eval()` function is deleted and replaced with `eval_body()` now.

### CODE REVIEW

**Review by aviatesk — APPROVED** [review]

**Comment by aviatesk** [comment]
Thanks for taking a look at this.
I found we may also want to update another part of this documentation on this opportunity ?

What do you think on this diff ?
```diff
diff --cc doc/src/devdocs/init.md
index dbd1938e25,2283131179..0000000000
--- a/doc/src/devdocs/init.md
+++ b/doc/src/devdocs/init.md
@@@ -158,19 -158,16 +158,16 @@@ executes it
  
  [`Base._start`](https://github.com/JuliaLang/julia/blob/master/base/client.jl) calls [`Base.process_options`](https://github.com/JuliaLang/julia/blob/master/base/client.jl)
  which calls [`jl_parse_input_line("println("Hello World!")")`](https://github.com/JuliaLang/julia/blob/master/src/ast.c)
- to create an expression object and [`Base.eval()`](@ref eval) to execute it.
+ to create an expression object and [`Core.eval(Main, ex)`](@ref Core.eval) to execute the parsed expression `ex` in the module context of `Main`.
  
- ## `Base.eval`
+ ## `Core.eval`
  
- [`Base.eval()`](@ref eval) was [mapped to `jl_f_top_eval`](https://github.com/JuliaLang/julia/blob/master/src/builtins.c)
- by `jl_init_primitives()`.
- 
- [`jl_f_top_eval()`](https://github.com/JuliaLang/julia/blob/master/src/builtins.c) calls [`jl_toplevel_eval_in(jl_main_module, ex)`](https://github.com/JuliaLang/julia/blob/master/src/builtins.c),
- where `ex` is the parsed expression `println("Hello World!")`.
- 
- [`jl_toplevel_eval_in()`](https://github.com/JuliaLang/julia/blob/master/src/builtins.c) calls
- [`jl_toplevel_eval_flex()`](https://github.com/JuliaLang/julia/blob/master/src/toplevel.c) which
- calls [`eval_body()` in `interpreter.c`](https://github.com/JuliaLang/julia/blob/master/src/interpreter.c).
+ [`Core.eval(Main, ex)`](@ref Core.eval) calls [`jl_toplevel_eval_in(m, ex)`](https://github.com/JuliaLang/julia/blob/master/src/toplevel.c),
+ which calls [`jl_toplevel_eval_flex`](https://github.com/JuliaLang/julia/blob/master/src/toplevel.c).
+ `jl_toplevel_eval_flex` implements a simple heuristic to decide whether to compile a given code thunk or run it by interpreter.
 -When given `println("Hello World!")`, it would usually decide to run it by interpreter,
 -in which case it calls [`jl_interpret_toplevel_thunk`](https://github.com/JuliaLang/julia/blob/master/src/interpreter.c),
 -which then calls [`eval_body`](https://github.com/JuliaLang/julia/blob/master/src/interpreter.c).
++When given `println("Hello World!")`, it would usually decide to run the code by interpreter, in which case it calls
++[`jl_interpret_toplevel_thunk`](https://github.com/JuliaLang/julia/blob/master/src/interpreter.c), which then calls
++[`eval_body`](https://github.com/JuliaLang/julia/blob/master/src/interpreter.c).
  
  The stack dump below shows how the interpreter works its way through various methods of [`Base.println()`](@ref)
  and [`Base.print()`](@ref) before arriving at [`write(s::IO, a::Array{T}) where T`](https://github.com/JuliaLang/julia/blob/master/base/stream.jl)
@@@ -209,12 -206,12 +206,11 @@@ Hello World
  | `jl_apply_generic()`           | `gf.c`          | `Base.println(String,)`                              |
  | `jl_apply()`                   | `julia.h`       |                                                      |
  | `do_call()`                    | `interpreter.c` |                                                      |
 -| `eval()`                       | `interpreter.c` |                                                      |
 -| `jl_interpret_toplevel_expr()` | `interpreter.c` |                                                      |
 -| `jl_toplevel_eval_flex()`      | `toplevel.c`    |                                                      |
 -| `jl_toplevel_eval()`           | `toplevel.c`    |                                                      |
 -| `jl_toplevel_eval_in()`        | `builtins.c`    |                                                      |
 -| `jl_f_top_eval()`              | `builtins.c`    |                                                      |
 +| `eval_body()`                  | `interpreter.c` |                                                      |
- | `jl_interpret_toplevel_expr()` | `interpreter.c` |                                                      |
- | `jl_toplevel_eval_flex()`      | `toplevel.c`    |                                                      |
- | `jl_toplevel_eval()`           | `toplevel.c`    |                                                      |
- | `jl_toplevel_eval_in()`        | `builtins.c`    |                                                      |
- | `jl_f_top_eval()`              | `builtins.c`    |                                                      |
++| `jl_interpret_toplevel_thunk`  | `interpreter.c` |                                                      |
++| `jl_toplevel_eval_flex`        | `toplevel.c`    |                                                      |
++| `jl_toplevel_eval_in`          | `builtins.c`    |                                                      |
++| `eval`                         | `boot.jl`       |                                                      |
  
  Since our example has just one function call, which has done its job of printing "Hello World!",
  the stack now rapidly unwinds back to `main()`.
```
(we may want to verify this with the actual stacktrace)

We can just merge this PR as is if you don't feel willing to update this document further though.

**Comment by akawashiro** [comment]
Your change looks good. Thank you. Should I commit myself?

By the way, `Base.process_options` is also outdated. `exec_options` is used insted.

**Comment by aviatesk** [comment]
You can add me as [co-author](https://stackoverflow.com/questions/7442112/how-to-attribute-a-single-commit-to-multiple-developers) if you want. I'm fine with either.

> By the way, Base.process_options is also outdated. exec_options is used insted.

Yes, let's update any other findings as well (Documentation is never up-to-date :p).

**Comment by akawashiro** [comment]
Thank you. I have done it.

File: doc/src/devdocs/init.md [hunk 1]
```diff
@@ -209,12 +206,11 @@ Hello World!
...
-| `eval()`                       | `interpreter.c` |                                                      |
-| `jl_interpret_toplevel_expr()` | `interpreter.c` |                                                      |
-| `jl_toplevel_eval_flex()`      | `toplevel.c`    |                                                      |
-| `jl_toplevel_eval()`           | `toplevel.c`    |                                                      |
-| `jl_toplevel_eval_in()`        | `builtins.c`    |                                                      |
-| `jl_f_top_eval()`              | `builtins.c`    |                                                      |
+| `eval_body()`                  | `interpreter.c` |                                                      |
+| `jl_interpret_toplevel_thunk`  | `interpreter.c` |                                                      |
+| `jl_toplevel_eval_flex`        | `toplevel.c`    |                                                      |
+| `jl_toplevel_eval_in`          | `builtins.c`    |                                                      |
```
Comment (aviatesk) [inline doc/src/devdocs/init.md:212]:
```suggestion
| `jl_toplevel_eval_in`          | `toplevel.c`    |                                                      |
```

File: doc/src/devdocs/init.md [hunk 2]
```diff
@@ -156,21 +156,18 @@ executes it.
 
 ## `Base._start`
 
-[`Base._start`](https://github.com/JuliaLang/julia/blob/master/base/client.jl) calls [`Base.process_options`](https://github.com/JuliaLang/julia/blob/master/base/client.jl)
+[`Base._start`](https://github.com/JuliaLang/julia/blob/master/base/client.jl) calls [`Base.exec_options`](https://github.com/JuliaLang/julia/blob/master/base/client.jl)
 which calls [`jl_parse_input_line("println("Hello World!")")`](https://github.com/JuliaLang/julia/blob/master/src/ast.c)
-to create an expression object and [`Base.eval()`](@ref eval) to execute it.
+to create an expression object and [`Core.eval(Main, ex)`](@ref Core.eval) to execute the parsed expression `ex` in the module context of `Main`.
```
Comment (aviatesk) [inline doc/src/devdocs/init.md:161]:
```suggestion
to create an expression object and [`Core.eval(Main, ex)`](@ref eval) to execute the parsed expression `ex` in the module context of `Main`.
```

File: doc/src/devdocs/init.md [hunk 3]
```diff
@@ -156,21 +156,18 @@ executes it.
...
-[`Base.eval()`](@ref eval) was [mapped to `jl_f_top_eval`](https://github.com/JuliaLang/julia/blob/master/src/builtins.c)
-by `jl_init_primitives()`.
-
-[`jl_f_top_eval()`](https://github.com/JuliaLang/julia/blob/master/src/builtins.c) calls [`jl_toplevel_eval_in(jl_main_module, ex)`](https://github.com/JuliaLang/julia/blob/master/src/builtins.c),
-where `ex` is the parsed expression `println("Hello World!")`.
-
-[`jl_toplevel_eval_in()`](https://github.com/JuliaLang/julia/blob/master/src/builtins.c) calls
-[`jl_toplevel_eval_flex()`](https://github.com/JuliaLang/julia/blob/master/src/toplevel.c) which
-calls [`eval()` in `interpreter.c`](https://github.com/JuliaLang/julia/blob/master/src/interpreter.c).
+[`Core.eval(Main, ex)`](@ref Core.eval) calls [`jl_toplevel_eval_in(m, ex)`](https://github.com/JuliaLang/julia/blob/master/src/toplevel.c),
```
Comment (aviatesk) [inline doc/src/devdocs/init.md:165]:
```suggestion
[`Core.eval(Main, ex)`](@ref eval) calls [`jl_toplevel_eval_in(m, ex)`](https://github.com/JuliaLang/julia/blob/master/src/toplevel.c),
```

**Review by aviatesk — COMMENTED** [review]
This documentation syntax broke the whole doctest system ... ?

File: doc/src/devdocs/init.md [hunk 4]
```diff
@@ -156,21 +156,18 @@ executes it.
...
 ## `Base._start`
 
-[`Base._start`](https://github.com/JuliaLang/julia/blob/master/base/client.jl) calls [`Base.process_options`](https://github.com/JuliaLang/julia/blob/master/base/client.jl)
+[`Base._start`](https://github.com/JuliaLang/julia/blob/master/base/client.jl) calls [`Base.exec_options`](https://github.com/JuliaLang/julia/blob/master/base/client.jl)
 which calls [`jl_parse_input_line("println("Hello World!")")`](https://github.com/JuliaLang/julia/blob/master/src/ast.c)
-to create an expression object and [`Base.eval()`](@ref eval) to execute it.
+to create an expression object and [`Core.eval(Main, ex)`](@ref Core.eval) to execute the parsed expression `ex` in the module context of `Main`.
 
-## `Base.eval`
+## `Core.eval`
```
Comment (aviatesk) [inline doc/src/devdocs/init.md:163]: FWIW, the previous `Base.eval` section creates a false documentation binding `@ref Base.eval`, which are being used in several other places, which broke the documentation system on this PR.

They should point to `Base.MainInclude.eval` instead, so I fixed all the references.

Once the CI passes, this should be good to go.

Comment (akawashiro) [reply doc/src/devdocs/init.md:163]: Thank you. It is interesting that CI checks documents also.

File: base/docs/basedocs.jl [hunk 5]
```diff
@@ -130,7 +130,7 @@ kw"__init__"
     baremodule
 
 `baremodule` declares a module that does not contain `using Base` or local definitions of
-[`eval`](@ref Base.eval) and [`include`](@ref Base.include). It does still import `Core`. In other words,
+[`eval`](@ref Base.Main.include) and [`include`](@ref Base.include). It does still import `Core`. In other words,
```
Comment (aviatesk) [inline base/docs/basedocs.jl:133]:
```suggestion
[`eval`](@ref Base.MainInclude.eval) and [`include`](@ref Base.include). It does still import `Core`. In other words,
```

**Review by aviatesk — APPROVED** [review]

**Comment by akawashiro** [comment]
Thank you.

**Comment by aviatesk** [comment]
Much appreciated !

**Pull request merged**